### PR TITLE
[New] Sketch on map

### PR DESCRIPTION
**Pull request by rolson**
## Description

This PR implements `Sketch on map` in `Display information` category.
URL to README [here](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/9953133126090fb2ce92a932245bf5d1376d982e/Shared/Samples/Sketch%20on%20the%20map/README.md)

## Linked Issue(s)

- `swift/issues/3580`

## How To Test

- Tap the pencil menu button and choose a tool for interactive editing
- Tap the pencil again for interactive editing options including undo/redo/delete/clear/save, etc

## Screenshots

sketch-on-map-1
sketch-on-map-2

### CODE REVIEW

File: Samples.xcodeproj/project.pbxproj
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:1117]: This change can be reverted.

Comment (rolson) [reply Samples.xcodeproj/project.pbxproj:1117]:
```suggestion
				minimumVersion = 200.0.0-beta;
```

Comment (rolson) [reply Samples.xcodeproj/project.pbxproj]: I reverted that change, but Xcode keeps making it, so you might want to go with it.

**Comment by rolson** [comment]
@yo1995 - this is still ready for review but I did make a commit that simplifies the logic and also fixes a bug with the model.

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (dfeinzimer) [inline Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:22]:
```suggestion
    @State private var map = Map(basemapStyle: .arcGISTopographic)
    
    /// The model that is required by the menu.
```

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (dfeinzimer) [inline Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:255]:
```suggestion
    /// - Parameter geometry: The geometry of the graphic to be saved.
    /// - Returns: Either a marker or fill symbol depending on the type of provided geometry.
    private func symbol(for geometry: Geometry) -> Symbol {
```
Or any return statement.

File: Shared/Samples/Sketch on the map/README.md
Comment (yo1995) [inline Shared/Samples/Sketch on the map/README.md:1]: The common design changed the name to `Sketch on map` without "the". Please rename

- the folder
- README title
- title in metadata
- swift filename
- view type name

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (yo1995) [inline Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:1]:
```suggestion
// Copyright 2023 Esri
```

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (dfeinzimer) [inline Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:78]: For some reason, `pencil.line` doesn't render for me. 🤔 

Screenshot 2023-03-22 at 17 17 40

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (yo1995) [inline Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:232]: Do you mean to show it in the docstring as a precondition?
```suggestion
    /// - Precondition: `canSave`
```

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (yo1995) [inline Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:241]: I'm wondering if the `save()` method above has a precondition, this can also have 

`/// - Precondition: canClearSavedSketches`

**Review by yo1995 — COMMENTED** [review]
The code is quite clean. I like the creative usage of SFSymbols in the menu. 

After the sample name is changed, I'll send you the screenshots via Slack.

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (rolson) [reply Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:78]: I would recommend cleaning, deleting derived data and trying again.

Can you try some other symbols and see if they exist?

I probably wouldn't hold up the PR for this.

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (rolson) [reply Shared/Samples/Sketch on the map/SketchOnTheMapView.swift:242]: it's not necessary in the same way.  The save method does a force-unwrap of the geometry and that's why the precondition is necessary.

**Review by yo1995 — COMMENTED** [review]
Please also resolve the merge conflict.

**Comment by yo1995** [comment]
The rename in 7451cbce1a513d1f925f82357c57bdddeaf95881 missed the files in the project file: 
image
You can remove the reference and add them back again.

File: Shared/Samples/Sketch on map/README.metadata.json
Comment (yo1995) [inline Shared/Samples/Sketch on map/README.metadata.json:30]:
```suggestion
        "SketchOnMapView.swift"
```

File: Shared/Samples/Sketch on the map/SketchOnTheMapView.swift
Comment (yo1995) [reply Shared/Samples/Sketch on the map/SketchOnTheMapView.swift]: image

I'm guessing it is related to SF Symbols versions. On iPhone 14 Pro Max iOS 16.2 sim I see it display, whereas it doesn't show on iPhone 11 iOS 15.5. Maybe it is a new symbol shipped with a later version of iOS.

Comment (yo1995) [reply Shared/Samples/Sketch on the map/SketchOnTheMapView.swift]: So if we target iOS 15.0, an alternative can be `line.diagonal`

Comment (dfeinzimer) [reply Shared/Samples/Sketch on the map/SketchOnTheMapView.swift]: Yup, I think this is correct, I'm seeing the same.

File: Samples.xcodeproj/project.pbxproj
Comment (rolson) [inline Samples.xcodeproj/project.pbxproj:63]: I'm not sure why these are showing up as diffs.  They were in the v.next branch when I fixed the merge conflict.  Any idea?

Comment (yo1995) [reply Samples.xcodeproj/project.pbxproj:63]: Sometimes when people use the GitHub editor to do "interactive" merge conflict resolve, the order of the files can be messed up and Xcode isn't happy about them. Xcode will try to make auto-corrections whenever a new change comes in. I'm OK to leave them in this PR even thought they aren't related.

**Review by dfeinzimer — APPROVED** [review]
Good to go apart from the missing "New Line" symbol in older sims.

File: Shared/Samples/Sketch on map/SketchOnMapView.swift
Comment (rolson) [inline Shared/Samples/Sketch on map/SketchOnMapView.swift:79]: updated to use more available symbol- @dfeinzimer

**Review by dfeinzimer — DISMISSED** [review]

**Review by yo1995 — DISMISSED** [review]

**Comment by rolson** [comment]
please re-approve.  I changed the "New Point" icon

**Comment by rolson** [comment]
hold up, another merge conflict

**Comment by yo1995** [comment]
Because Mark merged #128 just now, you'll need to resolve conflicts one more time.

**Comment by rolson** [comment]
I'm seeing this difference from the latest v.next:
image
@mhdostal - did you inadvertently remove the oauth sample group?

**Review by yo1995 — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Comment by yo1995** [comment]
> did you inadvertently remove the oauth sample group?

I don't think so - the diff shows the change background sample is on `v.next`, and the oauth change is on your branch. I'll do some project file housekeeping after more samples are merged.

**Comment by rolson** [comment]
> did you inadvertently remove the oauth sample group?

>> I don't think so - the diff shows the change background sample is on v.next, and the oauth change is on your branch. I'll do some project file housekeeping after more samples are merged.

I think actually it was removed sometime before Mark's merge.

**Comment by yo1995** [comment]
Hi Ryan, would you make an update once `swift/pull/3811` is merged? Thanks 😁

**Comment by mhdostal** [comment]
> @mhdostal - did you inadvertently remove the oauth sample group?

That is entirely possible.  Hand-editing project files is tricky.  Is everything OK now?

**Comment by rolson** [comment]
> Hi Ryan, would you make an update once swift/pull/3811 is merged? Thanks 😁

I've asked @philium to do that, thanks

**Pull request merged**
